### PR TITLE
feat(query-builder): Focus filter when pressing backspace and value input is empty

### PR DESCRIPTION
**Pull request by malwilley**
When editing a tag value, pressing backspace with an empty input used to do nothing. This modifies the behavior to focus the whole token, allowing easier deletion if you change your mind while creating a filter.

https://github.com/getsentry/sentry/assets/10888943/4d31a032-eeed-4b83-ad91-307e25f19655

### CODE REVIEW

File: static/app/components/searchQueryBuilder/filter.tsx [hunk 1]
```diff
@@ -23,11 +23,15 @@ import {t} from 'sentry/locale';
 import {space} from 'sentry/styles/space';
 import {defined} from 'sentry/utils';
 
-type SearchQueryTokenProps = {
+interface SearchQueryTokenProps {
```
Comment (leeandher) [inline static/app/components/searchQueryBuilder/filter.tsx:26]: ❤️

File: static/app/components/searchQueryBuilder/valueCombobox.tsx [hunk 2]
```diff
@@ -776,16 +778,11 @@ export function SearchQueryBuilderValueCombobox({
       }
 
       // If at the start of the input and backspace is pressed, delete the last selected value
```
Comment (leeandher) [inline static/app/components/searchQueryBuilder/valueCombobox.tsx:780]: I don't think this comment is accurate anymore

File: static/app/components/searchQueryBuilder/valueCombobox.tsx [hunk 3]
```diff
@@ -776,16 +778,11 @@ export function SearchQueryBuilderValueCombobox({
...
-      if (
-        e.key === 'Backspace' &&
-        e.currentTarget.selectionStart === 0 &&
-        e.currentTarget.selectionEnd === 0 &&
-        canSelectMultipleValues
-      ) {
-        dispatch({type: 'DELETE_LAST_MULTI_SELECT_FILTER_VALUE', token});
```
Comment (leeandher) [inline static/app/components/searchQueryBuilder/valueCombobox.tsx:785]: I'm not the most familiar with the new search query code, but Is it expected that this action isn't being dispatched anymore? I see onDelete was modified but this call isn't in it's new definition.

File: static/app/components/searchQueryBuilder/valueCombobox.tsx [hunk 4]
```diff
@@ -440,17 +441,17 @@ function cleanFilterValue(key: string, value: string): string {
...
     case FieldValueType.INTEGER:
       if (FILTER_VALUE_INT.test(value)) {
         return value;
       }
-      return '';
+      return null;
     case FieldValueType.DATE:
       const parsed = parseFilterValueDate(value);
 
       if (!parsed) {
```
Comment (leeandher) [inline static/app/components/searchQueryBuilder/valueCombobox.tsx:453]: super nit: parsed is only used once so we can put it in the conditional

Comment (leeandher) [reply static/app/components/searchQueryBuilder/valueCombobox.tsx:453]: also  🙌  to swapping `''` for `null` to be more explicit about unmatched values.

**Review by leeandher — APPROVED** [review]
One question, rest looks good though! so many quality of life features 💪

File: static/app/components/searchQueryBuilder/valueCombobox.tsx (hunk 2 above)
Comment (malwilley) [reply static/app/components/searchQueryBuilder/valueCombobox.tsx:780]: You are correct!

**Pull request merged**